File: solution/src/oblig1/InsertionSort.java
package oblig1;

import java.io.BufferedReader;
import java.io.InputStreamReader;

class InsertionSort {

    public static void main(String[] args) {
        // Read input
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        int[] A = input.lines().mapToInt(i -> Integer.parseInt(i)).toArray();

        insertionSort(A);

        // Print result
        for (int num : A) {
            System.out.println(num);
        }
    }

    static void insertionSort(int[] A) {
        // Skriv her
    }
}

File: solution/src/oblig1/insertionsort.py
import sys


def main():
    # Read input
    A = [int(line) for line in sys.stdin]

    insertionsort(A)

    # Print result
    for num in A:
        print(num)


def insertionsort(A):
    for i in range(1, len(A)):
        j = i
        while j > 0 and A[j - 1] > A[j]:
           A[j - 1], A[j] = A[j], A[j - 1]
           j -= 1


if __name__ == "__main__":
    main()
